include/AssembleElemSolverAlgorithm.h

```diff
@@ -87,7 +87,7 @@ class AssembleElemSolverAlgorithm : public SolverAlgorithm
         auto bktId = elem_buckets.device_get(team.league_rank());
         auto& b = ngpMesh.get_bucket(entityRank, bktId);
 
-#ifndef KOKKOS_ENABLE_CUDA
+#if !defined(KOKKOS_ENABLE_CUDA) && !defined(KOKKOS_ENABLE_HIP)
```

Comment:
I think we need to wrap this into a single macro and have something like `NALU_USING_GPU` or `NALU_USING_NGP` defined in a top level header like `NgpTypes.h` or something like that.